tests/acceptance/features/bootstrap/SettingsContext.php

```diff
@@ -238,10 +236,14 @@ public function userGetAssignmentsList(string $user): void {
 	 */
 	public function userShouldHaveRole(string $user, string $role): void {
 		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
-		$this->sendRequestAssignmentsList($this->featureContext->getAdminUserName(), $userId);
-		$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
-		$assignmentRoleId = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignments"][0]["roleId"];
-		Assert::assertEquals($this->userGetRoleIdByRoleName($this->featureContext->getAdminUserName(), $role), $assignmentRoleId, "user $user has no role $role");
+		$response = $this->sendRequestAssignmentsList($this->featureContext->getAdminUserName(), $userId);
+		$assignmentResponse = $this->featureContext->getJsonDecodedResponseBodyContent($response);
+		if (isset($assignmentResponse->assignments[0]->roleId)) {
+			$assignmentRoleId = $assignmentResponse->assignments[0]->roleId;
+			Assert::assertEquals($this->userGetRoleIdByRoleName($this->featureContext->getAdminUserName(), $role), $assignmentRoleId, "user $user has no role $role");
```

Comment:
```suggestion
			$actualRoleId = $assignmentResponse->assignments[0]->roleId;```suggestion
                        $expectedRoleId = $this->userGetRoleIdByRoleName($this->featureContext->getAdminUserName(), $role);
			Assert::assertEquals($expectedRoleId, $actualRoleId, "user '$user' has no role '$role'");
```